kubernetes/shuffle-and-export.sh

```diff
@@ -1,14 +1,62 @@
-#!/bin/bash -e
-cd /engines/KataGo-custom/python
+#!/bin/bash -eu
+
+USE_GATING=0
+# Command line flag parsing (https://stackoverflow.com/a/33826763/4865149).
+# Flags must be specified before positional arguments.
+while [ -n "${1-}" ]; do
+  case $1 in
+    # Set this flag if the gatekeeper is enabled.
+    --gating) USE_GATING=1 ;;
+    # Pre-seed with this training data as the source.
+    # If the pre-seed source directory is formatted as `*/selfplay/t0-s*-d*`,
+    # then all earlier directories with lower step count in `*/selfplay/` will
+    # also be included in the pre-seeding
+    --preseed) PRESEED_SRC=$2; shift ;;
+    -*) echo "Unknown parameter passed: $1"; usage; exit 1 ;;
+    *) break ;;
+  esac
+  shift
+done
+
 RUN_NAME="$1"
 DIRECTORY="$2"
 VOLUME_NAME="$3"
-# 1 to enable gatekeeper, 0 to disable gatekeeper
-USE_GATING="$4"
+shift
+shift
+shift
 
 # not related to shuffle-and-export but we want some process to log this
 /go_attack/kubernetes/log-git-commit.sh /"$VOLUME_NAME"/victimplay/"$DIRECTORY"
 
-mkdir -p /"$VOLUME_NAME"/victimplay/"$DIRECTORY"
-./selfplay/shuffle_and_export_loop.sh    "$RUN_NAME"    /"$VOLUME_NAME"/victimplay/"$DIRECTORY"    /tmp    16    256    $USE_GATING
+EXPERIMENT_DIR=/"$VOLUME_NAME"/victimplay/"$DIRECTORY"
+mkdir --parents "$EXPERIMENT_DIR"
+
+PRESEED_DST="$EXPERIMENT_DIR"/selfplay/prev-selfplay
+if [ -n "${PRESEED_SRC:-}" ] && [ ! -d "$PRESEED_DST" ]; then
+  PRESEED_SRC=$(realpath "$PRESEED_SRC")
+  if [[ "$PRESEED_SRC" =~ -s([0-9]+)-d[0-9]+ ]]; then
+    # Preseed data up to the step count listed in PRESEED_SRC.
+    FINAL_STEP=${BASH_REMATCH[1]}
+    mkdir --parents "$PRESEED_DST"
+    PRESEED_SRC_PARENT=$(dirname "$PRESEED_SRC")
+    for DIR in "$PRESEED_SRC_PARENT"/*/; do
+      DIR_NAME=$(basename "$DIR")
+      if [ "$DIR_NAME" = "prev-selfplay" ] || [ "$DIR_NAME" = "random" ]; then
+        ln -s "$DIR" "$PRESEED_DST"
+      elif [[ "$DIR" =~ -s([0-9]+)-d[0-9]+ ]]; then
+        STEP=${BASH_REMATCH[1]}
+        if [ "$STEP" -le "$FINAL_STEP" ]; then
+          ln -s "$DIR" "$PRESEED_DST"
+        fi
+      else
+        echo "Skipping unrecognized pre-seed source: $DIR"
+      fi
+    done
+  else
+    ln -s "$PRESEED_SRC" "$PRESEED_DST"
+  fi
+fi
+
+cd /engines/KataGo-custom/python
+./selfplay/shuffle_and_export_loop.sh "$RUN_NAME" "$EXPERIMENT_DIR" /tmp 16 256 $USE_GATING $@
```

Comment:
Quotes needed to avoid mangling arguments: https://stackoverflow.com/questions/4824590/propagate-all-arguments-in-a-bash-shell-script
```suggestion
./selfplay/shuffle_and_export_loop.sh "$RUN_NAME" "$EXPERIMENT_DIR" /tmp 16 256 $USE_GATING "$@"
```